task6/posts.js

```diff
@@ -0,0 +1,744 @@
+            return false;
+        }
+        if (post.description.length > 200)
+            return false;
+        for (let str of post.likes) if (typeof(str) !== 'string')
+            return false;
+        for (let str of post.hashTags) if (typeof(str) !== 'string')
+            return false;
+        return post.author !== '';
+    }
+
+    function addPost(post) {
+        if (typeof post !== 'object') {
+            return false;
+        }
+        if (!validatePost(post)) {
+            console.log('Invalid post');
+            return false;
+        }
+        if (typeof(getPost(post.id)) !== 'undefined') {
+            return false;
+        }
+        posts.push(post);
+        return true;
+    }
+
+    function editPost(id, post) {
+        if (typeof post !== 'object') {
+            console.log('Invalid argument: "post" isn\'t object');
+            return false;
+        }
+        if (id === '') {
+            console.log('Invalid argument: "id" is empty');
+            return false;
+        }
+        let source = getPost(id);
+        if (typeof source === 'undefined') {
+            console.log(`Wrong argument: post with id = '${id}' does not exist`);
+            return false;
+        }
+        for (let key in post)
+            if (!(key in source)) {
+                console.log(`Invalid argument: "post" contains invalid property '${key}'`);
+                return false;
+            }
+        let copy = {};
+        Object.assign(copy, source);
+        for (let key in post) {
+            if (!['id', 'author', 'createdAt', 'likes', 'hashTags'].includes(key))
+                copy[key] = post[key];
+        }
+        if ('likes' in post)
+            copy['likes'] = post.likes.slice();
+        if ('hashTags' in post)
+            copy['hashTags'] = post.hashTags.slice();
+        copy.id = 'falseId';
+        if (validatePost(copy)) {
+            copy.id = id;
+            Object.assign(source, copy);
+        }
+        else return false;
+        return true;
+    }
+
+    function removePost(id) {
+        if (id === '') {
+            console.log('Invalid argument: "id" is empty');
+            return;
+        }
+        let ind = -1;
+        let source = posts.find((item, index) => {
+            ind = index;
+            return item.id === id;
+        });
+        if (typeof source === 'undefined') {
+            console.log(`Wrong argument: post with id = '${id}' does not exist`);
+            return false;
+        }
+        posts.splice(ind, 1);
+        return true;
+    }
+
+    function addLike(id, user) {
+        if(typeof(id) !== "string" || typeof(user) !== "string")
+            return false;
+
+        let post = getPost(id);
+
+        if(typeof(post) === "undefined")
+            return false;
+
+        if(post.likes.includes(user))
+            return false;
+
+        post.likes.push(user);
+
+        return true;
+    }
+
+    function removeLike(id, user) {
```

Comment:
According to that case (furthermore, code is almost similar), it's better to have smth like `toggleLike`, that adds like if it wasn't there or removes it otherwise.